hypothesis-python/src/hypothesis/internal/conjecture/engine.py

```diff
@@ -1114,19 +1155,48 @@ def uniform(random, n):
     return int_to_bytes(random.getrandbits(n * 8), n)
 
 
+def pop_random(random, values):
+    i = random.randrange(0, len(values))
+    result = values[i]
+    end = values.pop()
+    if i < len(values):
+        values[i] = end
+    return result
```

Comment:
minor: `pop()` takes an optional argument, the index of the item to remove. I think the following code is equivalent, but preserves the original order for the list, and it’s shorter:

```python
def pop_random(random, values):
    idx = random.randrange(0, len(values))
    return values.pop(idx)
```

But maybe removing from the middle of a list is more expensive?

Reply:
Yes, the reason for doing this way is that it's an `O(1)` operation and the order of the list doesn't matter because we only access its elements through random sampling.

Reply:
I've added a comment explaining this to the function.